sympy/concrete/summations.py

```diff
@@ -253,6 +257,174 @@ def _eval_simplify(self, ratio, measure):
     def _eval_summation(self, f, x):
         return None
 
+    def is_convergent(self):
+        """
+        Convergence tests are used for checking the convergence of
+        a series. There are various tests employed to check the
+        convergence, returns True if convergent and false if divergent and
+        None if unknow. Like divergence test, root test, integral test,
+        alternating series test, comparison tests.
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Convergence_tests
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, factorial, S, Sum, Symbol, oo
+        >>> n = Symbol('n', integer=True)
+        >>> Sum(n/(n - 1), (n, 4, 7)).is_convergent()
+        True
+        >>> Sum(n/(2*n + 1), (n, 1, oo)).is_convergent()
+        False
+        >>> Sum(factorial(n)/5**n, (n, 1, oo)).is_convergent()
+        False
+        >>> Sum(1/n**(S(6)/5), (n, 1, oo)).is_convergent()
+        True
+        """
+        from sympy import Interval, Integral, Limit, log, singularities, symbols
+        from sympy.solvers.solveset import solveset
+        p, q = symbols('p q', cls=Wild)
+
+        sequence_term = self.function
+        sym = self.limits[0][0]
+        lower_limit = self.limits[0][1]
+        upper_limit = self.limits[0][2]
+        interval = Interval(lower_limit, upper_limit)
+
+        if lower_limit.is_finite and upper_limit.is_finite:
+            return True
+
+        # transform sym -> -sym and swap the upper_limit = S.Infinity and lower_limit = - upper_limit
+        if lower_limit is S.NegativeInfinity:
+            sequence_term = sequence_term.xreplace({sym: -sym})
+            lower_limit = -upper_limit
+            upper_limit = S.Infinity
+
+        ###  -------- Divergence test ----------- ###
+        try:
+            if limit(sequence_term, sym, upper_limit) != S.Zero:
+                return False
+        except NotImplementedError:
+            pass
+
+        order = O(sequence_term, (sym, S.Infinity))
+
+        ### --------- p-series test (1/n**p) ---------- ###
+        p1_series_test = order.expr.match(sym**p)
+        if not p1_series_test is None:
```

Comment:
I think the recommended form is ``if ... is not ...``. (No difference of meaning but more readable.)